radixdlt-core/radixdlt/src/main/java/com/radixdlt/middleware2/network/MessageCentralProxyPeerDiscovery.java

```diff
@@ -0,0 +1,98 @@
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.middleware2.network;
+
+import com.radixdlt.consensus.bft.BFTNode;
+import com.radixdlt.environment.rx.RemoteEvent;
+import com.radixdlt.network.messaging.MessageCentral;
+import com.radixdlt.network.p2p.NodeId;
+import com.radixdlt.network.p2p.discovery.ProxiedPeers;
+import com.radixdlt.network.p2p.discovery.messages.ProxiedPeersResponseMessage;
+import io.reactivex.rxjava3.core.BackpressureStrategy;
+import io.reactivex.rxjava3.core.Flowable;
+import java.util.Objects;
+import javax.inject.Inject;
+
+/** Network interface for peer discovery messages the MessageCentral */
+public final class MessageCentralProxyPeerDiscovery {
+  private final MessageCentral messageCentral;
+
+  @Inject
+  public MessageCentralProxyPeerDiscovery(MessageCentral messageCentral) {
+    this.messageCentral = Objects.requireNonNull(messageCentral);
+  }
+
+  public Flowable<RemoteEvent<ProxiedPeers>> proxiedPeersResponses() {
+    return this.messageCentral
+        .messagesOf(ProxiedPeersResponseMessage.class)
+        .toFlowable(BackpressureStrategy.BUFFER)
+        .map(m -> new RemoteEvent<>(m.sourceNode(), new ProxiedPeers(m.message().getPeers())));
+  }
+
+  public void sendProxiedPeersResponse(BFTNode node, ProxiedPeers peersResponse) {
+    this.messageCentral.send(
+        NodeId.fromBFTNode(node), new ProxiedPeersResponseMessage(peersResponse.peers()));
```

Comment:
let's rename `ProxiedPeersResponseMessage` to `ProxiedPeersMessage` for consistency

Reply:
Done